tools/maint/gen_struct_info.py

```diff
@@ -272,6 +279,35 @@ def inspect_headers(headers, cflags):
     cmd += ['-sMEMORY64=2', '-Wno-experimental']
 
   show(shared.shlex_join(cmd))
+  return cmd
+
+
+def inspect_headers(headers, cflags):
+  # Write the source code to a temporary file.
+  src_file = tempfile.mkstemp('.c', text=True)
+  show('Generating C code... ' + src_file[1])
+  code = generate_c_code(headers)
+  os.write(src_file[0], '\n'.join(code).encode())
+  os.close(src_file[0])
+  src_file_path = src_file[1]
```

Comment:
How about doing `src_file_fd, src_file_path = tempfile.mkstemp('.c', text=True)` above instead of adding this line?